Source/Scene/Camera.js

```diff
@@ -3230,9 +3240,35 @@ define([
         Matrix4.clone(camera._transform, result.transform);
         result._transformChanged = true;
 
+        if (defined(camera.cameraLimiter)) {
+            result.cameraLimiter = CameraLimiter.clone(camera.cameraLimiter, result.cameraLimiter);
+        }
+
         return result;
     };
 
+    var saveOrientation = new HeadingPitchRoll();
+    function limitPosition(camera) {
+        // following the convention from _adjustHeightForTerrain to maintain correct orientation
+        // before and after position is changed
+        if (defined(camera.cameraLimiter) && camera._mode === SceneMode.SCENE3D) {
+            // store camera's hpr values
+            saveOrientation.heading = camera.heading;
```

Comment:
I'm not sure this is needed.  I think `setView` already uses the current camera orientation if it's not specified

Reply:
it uses the current camera direction - when i tested both versions in the camera limiter tutorial it seems that not saving the previous orientation does not allow for rotations that are not centered around the globe - should this be the case though? [also the wording here's kind of weird so i can demo it for u if needed]

Reply:
Okay, I took a look at the code and you're correct, `setView` does not maintain HPR.  I think it used to, but we changed it so the behavior was the same as `flyTo`